aten/src/ATen/Context.h

```diff
@@ -114,12 +114,42 @@ class CAFFE2_API Context {
   void setDeterministicCuDNN(bool);
   bool deterministic() const;
   void setDeterministic(bool);
+
+  // Throws an error if `Context::deterministic()` is true. This method should be called by
```

Comment:
How about promoting this to a proper note:

```
# Note [Writing Nondeterministic Operations]
```

That way the comments can clearly refer to this note for more detail.

The note can then start by explaining the three things nondeterministic operations must do:

- Write a comment explaining why the operation is nondeterministic
- Add the call to alertNotDeterministic and the comment explaining that all nondeterministic operations should make this call
- Update the documentation of set_deterministic() in torch/__init__.py

It can even include an example, if you like

Reply:
(The note should probably include additional requirements, like when to call `alertCuBLASConfigNotDeterministic`, too.)